shell/platform/darwin/macos/framework/Source/FlutterEngineTest.mm

```diff
@@ -515,19 +516,38 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication* _Nonnu
     FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"test"
                                                         project:project
                                          allowHeadlessExecution:true];
+    weakEngine = engine;
+    binaryMessenger = engine.binaryMessenger;
+  }
 
-    // Get the binary messenger for the engine.
-    id<FlutterBinaryMessenger> binaryMessenger = engine.binaryMessenger;
-    ASSERT_TRUE([binaryMessenger isKindOfClass:[FlutterBinaryMessengerRelay class]]);
-    relay = (FlutterBinaryMessengerRelay*)binaryMessenger;
+  // Once the engine has been deallocated, verify the weak engine pointer is nil, and thus not
+  // retained by the relay.
+  EXPECT_NE(binaryMessenger, nil);
+  EXPECT_EQ(weakEngine, nil);
+}
 
-    // Verify the relay parent (the engine) is non-nil.
-    EXPECT_NE(relay.parent, nil);
+// Verify that the engine is not retained indirectly via the texture registry held by plugins.
+// Issue: https://github.com/flutter/flutter/issues/116445
+TEST_F(FlutterEngineTest, FlutterTextureRegistryDoesNotReturnEngine) {
+  __weak FlutterEngine* weakEngine;
+  id<FlutterTextureRegistry> textureRegistry;
```

Comment:
Shouldn't this be weak too?  Maybe you should add an assert that it it has a value other than nil at some point.  This test could pass if it is always `nil`.

Reply:
This should be strong since we want to guarantee that holding on to this doesn't retain the engine.

Reply:
Oh wait... it has that check already, right at the bottom, above where we check that the engine _is_ nil.

Reply:
I got my EQ and NE mixed up, lgtm!